src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHelpers.Unix.cs

```diff
@@ -113,18 +113,23 @@ internal static void CreateDirectory(string fullPath, UnixFileMode? mode, bool o
                 mode = ExtractPermissions;
             }
 
+            // Create missing parents with ExtractPermissions and add them to be set to
+            // DefaultDirectoryMode if they are not present in the archive later.
```

Comment:
Is it ok that for a moment the parent directories exist with incorrect permissions?

Reply:
Is it possible to flip the permissions so the directories are created with DefaultDirectoryMode and then changed to ExtractPermissions?

Reply:
> Is it ok that for a moment the parent directories exist with ExtractPermissions?

This is intentional. If we don't know the mode we use a minimal permission set to extract.

> Is it possible to flip the permissions so the directories are created with DefaultDirectoryMode and then changed to ExtractPermissions?

Later we either encounter the right permissions in the archive, which we then use.
Or we change it to `DefaultDirectoryMode` if we don't.

Reply:
I've simplied `TarHelpers.CreateDirectory` a bit. It creates unknown directories (that is: not yet encountered in the archive) now with default permissions. When I straced `tar` it also used default permissions, see https://github.com/dotnet/runtime/pull/72078#discussion_r921432953.

Reply:
>It creates unknown directories (that is: not yet encountered in the archive) now with default permissions

Yes, but when they are encountered, they should be updated to whatever `Mode`, says, no?

Reply:
The mode gets applied when overwrite is true.

Reply:
It should be applied also when its false.

Reply:
I just read your [other comment](https://github.com/dotnet/runtime/pull/74002#discussion_r948745356), if this is how `tar` works, then its fine. Although it's a bit strange.